src/components/Toast/Toast.tsx

```diff
@@ -1,93 +1,71 @@
-   */
-  dismissible?: boolean;
-  /**
-   * Name of toast SVG icon. Available options are: toast, error, warning-2, success
+   * The child node(s) contains the toast message. Note: the toast message is displayed inside a TextPassage, so children can contain raw HTML
    */
-  iconName?: string;
+  children: React.ReactNode;
   /**
-   * The title attribute for the SVG icon in the toast
+   * The color of the Toast, based on EDS defined colors. Also determines the icon used.
+   * Note that the Icon mapping matches the style of Banners.
    */
-  iconTitle?: string;
+  variant: Variant;
   /**
-   * Stylistic variations for the toast type.
-   * - **success** - results in a green toast
-   * - **warning** - results in a yellow toast
-   * - **error** - results in a red toast
-   * - **info** - results in a blue toast
+   * Callback when Toast is dismissed.
    */
-  variant?: 'success' | 'warning' | 'error' | 'info' | 'brand';
-}
+  onDismiss?: () => void;
+};
 
 /**
- * Primary UI component for user interaction
+ * ```ts
+ * import {Toast} from "@chanzuckerberg/eds-components";
+ * ```
+ *
+ * A toast used to provide information on the state of the page, usually in response to a
+ * user action. Ex: The user updates their profile, and a toast pop-up informs them that the
+ * data was successfully saved.
  */
 export const Toast = ({
-  iconTitle,
-  className,
   children,
-  iconName,
-  dismissible = true,
+  className,
   variant,
-  closeButtonText = 'Close',
+  onDismiss,
+  // Allow for additional attributes such as aria roles
   ...other
 }: Props) => {
   const componentClassName = clsx(
-    styles['toast'],
     className,
+    styles['toast'],
     variant === 'success' && styles['toast--success'],
-    variant === 'warning' && styles['toast--warning'],
-    variant === 'error' && styles['toast--error'],
-    variant === 'brand' && styles['toast--brand'],
+    variant === 'alert' && styles['toast--alert'],
   );
-  const [dismissed, setDismissed] = useState(false);
-
-  function onDismiss(e: any) {
-    e.preventDefault();
-    setDismissed(true);
-  }
-  return dismissed ? null : (
-    <div className={componentClassName} role="alert" {...other}>
-      {iconName && (
-        <Icon
-          name={iconName}
-          title={iconTitle}
-          className={styles['toast__icon']}
-        />
-      )}
-      <div className={styles['toast__body']}>{children}</div>
-
-      {dismissible && (
-        <Button
-          className={styles['toast__close-btn']}
-          variant="bare"
-          aria-label={closeButtonText}
-          iconName="close"
-          iconPosition="after"
-          onClick={(e: any) => onDismiss(e)}
-        />
-      )}
+  return (
+    <div className={componentClassName} {...other}>
+      <div className={styles['toast__content']}>
+        {/* TODO: point to internal NotificationIcon once fully migrated */}
+        <svg fill="currentColor" width="16px" height="16px">
```

Comment:
fyi my Icon change just landed so you can use `Icon` with all the expected props now 🙂